Allow switching between mph and kmph

dabreegster opened the pull request:
Fixes #409 

This may need some fine-tuning. Want to test it out @XioNoX?
<img width="2850" height="914" alt="image" src="https://github.com/user-attachments/assets/eb076007-277d-442f-b361-3def6c886723" />

XioNoX commented on web/src/common/colors.ts:
```diff
@@ -6,7 +6,8 @@ export let speedColorScale = [
   "#581845",
 ];
-export let speedLimits = [20, 30, 40, 50, 60, 70];
+export let speedLimitsMPH = [20, 30, 40, 50, 60, 70];
```
That's fine ! In France 40 and 60 are not common at all, but from https://wiki.openstreetmap.org/wiki/Key:maxspeed it's used in other countries.
I'm wondering if they should/could go up 100km/h for example. But I think for now it's fine as it's set here.

XioNoX replied:
Also `kph` is more correct than `kmph` but also `km/h` is more frequently used than `kph`.

dabreegster replied:
I switched to "k/ph" in all the user-visible text

XioNoX replied:
you mean km/h ?

dabreegster replied:
Yes, good catch :laughing:

XioNoX commented on web/src/context/ContextualLayers.svelte:
```diff
@@ -81,6 +82,15 @@
         <LanguagePicker />
 
+        <span style="font-size: 20px; margin-left: 8px; margin-top: 4px;">
+          Measurement units
```
Maybe we don't even need this text, as the choices are self explanatory.

XioNoX commented:
I finally managed to get it to work :)

The speed scale seems incorrect to me, especially the 65 and 95km/h marks are they're unlikely to match any real life speed limit.

<img width="437" height="360" alt="Screenshot From 2025-09-20 21-17-58" src="https://github.com/user-attachments/assets/aa51da23-0cb6-43c1-8b2e-1602a08eec51" />

dabreegster commented:
I switched that legend to put the labels underneath each colored rectangle (buckets), even though it's really limits. I'll try switching back (and adding an explicit entry for "more than 110".

But point taken about weird increments. What should they be? 6 equal increments of 10 would be 30, 40, 50, 60, 70, 80, which doesn't really go high enough. Increments of 15 also feel weird -- 30, 45, 60, 75, 90, 105.

We don't have to use 6 buckets either. The styling for mph and kmph can be totally different if that makes more sense.

XioNoX commented:
what do you think of :
* 10 km/h and under (usually pedestrian areas)
* 20 km/h (usually living streets)
* 30km/h (residential areas)
* 50km/h (main city roads)
* 70km/h and above (anything outside cities, or trunk roads in cities)

That's mostly for a city usage of LTN, I don't know if the tool is used outside of cities. But even there, not sure if there is a need to distinguish between any > 70km/h speed limit.

But looking at it differently (and that's probably a scope creep), maybe LTN should look at all the speed limits from the defined neighborhood and dynamically assign the colors ?
For example all the ones I'm working on are 5-50km/h max., and often just 1 (30) or 2 (30/50) max.

dabreegster commented:
I switched to the new suggested scale and like it, thanks :)

There was also a bug before -- speed was converted to MPH and stored rounded. So when we translate back to KPH, it was rounding up a little. I changed the backend to store the unrounded floating point, and now conversions behave better. I need to regenerate binary files for England and Scotland, and then I'll merge this PR. Thanks for the help working through it, and I'm happy to keep iterating on it if anything seems wrong.

>  maybe LTN should look at all the speed limits from the defined neighborhood and dynamically assign the colors ?

That sounds potentially surprising from a user perspective and definitely tougher to do, so I'll leave this for now, but interesting idea
